web/ui/react-app/src/pages/graph/DataTable.tsx

```diff
@@ -76,17 +79,21 @@ const DataTable: FC<QueryResult> = ({ data }) => {
       break;
     case 'matrix':
       rows = (limitSeries(data.result) as RangeSamples[]).map((s, index) => {
-        const valueText = s.values
-          .map((v) => {
-            return v[1] + ' @' + v[0];
-          })
-          .join('\n');
+        const valuesAndTimes = s.values.map((v) => {
+          const printedDatetime = moment.unix(v[0]).toISOString(useLocalTime);
+          return (
+            <React.Fragment>
+              {v[1]} @{<span title={printedDatetime}>{v[0]}</span>}
+              <br />
+            </React.Fragment>
+          );
+        });
         return (
-          <tr style={{ whiteSpace: 'pre' }} key={index}>
```

Comment:
We will need to keep the `whiteSpace: 'pre'` to show white space in things like label values correctly. For example, imagine a label value with multiple spaces in a row, which would otherwise be shown as a single space in HTML.

To simulate this:

    absent(foo{bar="multiple    spaces"})[1m:]